### ISSUE TITLE
MySQL range

### ISSUE THREAD

**Issue by micheleselea** [pocoproject/poco#2110]
Suppose you have MYSQL DB and you want to copy data to another SQLITE ram DB. Fetching records, ordered by datetime, from MySQL db using RecordSet and range expression seems to have performance issue that making SQL query directly with LIMIT OFFSET,VALUE does not have
#### Expected behavior
Making 

```
int iLen=20000;
int iOffset=0;
do{
Poco::Data::Statement select(dbSession);
select <<  Poco::format("SELECT * FROM car_plates WHERE (DateTime<=%Ld) ORDER BY DateTime DESC LIMIT %Ld,%Ld",iMaxDateTime,iOffset,iLen);
select.execute();
Poco::Data::RecordSet rs(select);
bool more = rs.moveFirst();
if (!more)
       break;
//do stuff
iOffset+=iLen;
}while(bSomethingToDo);
```

and

```
int iLen=20000;
int iOffset=0;
Poco::Data::Statement select(dbSession);
select <<  Poco::format("SELECT * FROM car_plates WHERE (DateTime<=%Ld) ORDER BY DateTime DESC",iMaxDateTime),Keywords::range(0,iLen);
while (!select.done()) {
select.execute();
Poco::Data::RecordSet rs(select);
bool more = rs.moveFirst();
if (!more)
       break;
//do stuff
};
```

Should we have same performance? or I have misunderstand range scope?
#### Actual behavior
using Keywords::range(0,iLen); seems to make execute slower from mysql disk access point of view.

#### POCO version
1.8.1
#### Compiler and version
VS2015
#### Operating system and version
windows 10

**Comment by aleks-f** [comment]
Poco::Data semantics happen at the library level, so it will select the whole thing every time and then only extract the desired range out of it. MySQL, of course, selects only what you asked for in a step. We do not get involved in SQL query optimizations because doing it on the abstract Data library level would open a can of worms with SQL dialects, so it is up to the user to design optimal query and the surrounding algorithm to get the data out.

Bottom line - what you are seeing is expected behavior. Data library, and especially the convenience classes like RecordSet, are not meant for optimal performance in high data volume scenarios.

**Comment by micheleselea** [comment]
Ok understand,,,so I was misunderstood the functionality: I thought range was a generalization of LIMIT statement.
